.pre-commit-config.yaml

```diff
@@ -18,3 +18,14 @@ repos:
     hooks:
       - id: isort
         args: [ "--profile", "black"]
+
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.3.0
+    hooks:
+      - id: check-yaml
+        args: [--unsafe]
+      - id: end-of-file-fixer
```

Comment:
does this remove newline at EOF or does it add if missing? we're in the add newline at EOF camp

Reply:
It checks whether a new line has been added at EOF and if it's not then adds it. Also if a file has more than one new line then it removes extra ones.